include/xrpl/protocol/SField.h

```diff
@@ -346,7 +376,8 @@ using SF_UINT384 = TypedField<STBitString<384>>;
 using SF_UINT512 = TypedField<STBitString<512>>;
 
 using SF_ACCOUNT = TypedField<STAccount>;
-using SF_AMOUNT = TypedField<STAmount>;
+using SF_AMOUNT = TypedFieldAmount<SFieldMPT::No>;
+using SF_EITHER_AMOUNT = TypedFieldAmount<SFieldMPT::Yes>;
```

Comment:
This seems misleading to me. The template parameter controls whether the type can contain an mpt, but it doesn't necessarily need to hold an mpt - it can still hold an stamount. Maybe rename "Yes" to something else? `MaybeMPT` or somesuch?